kaldb/src/main/java/com/slack/kaldb/metadata/dataset/DatasetMetadata.java

```diff
@@ -32,11 +34,22 @@ public DatasetMetadata(
     checkArgument(throughputBytes >= 0, "throughputBytes must be greater than or equal to 0");
     checkPartitions(partitionConfigs, "partitionConfigs must not overlap start and end times");
 
+    checkArgument(
+        serviceNamePattern.length() <= 256, "serviceNamePattern must be no longer than 256 chars");
+    checkArgument(
```

Comment:
We probably don't need this check here - as long as it's not null / empty that would be a valid pattern; this will especially be the case if we move to glob/regex.

If this is indeed a regex pattern in the future, we probably would want to compile that pattern and store it in this class once, instead of at runtime.